passes/proc/proc_rmdead.cc

```diff
@@ -28,9 +28,62 @@
 USING_YOSYS_NAMESPACE
 PRIVATE_NAMESPACE_BEGIN
 
-void proc_rmdead(RTLIL::SwitchRule *sw, int &counter, int &full_case_counter)
+static bool can_use_explicit_const_pool(RTLIL::SwitchRule *sw)
+{
+	if (!GetSize(sw->signal))
+		return false;
+
+	for (const RTLIL::SigBit &bit : sw->signal)
+		if (bit.wire == NULL)
+			return false;
+
+	for (const RTLIL::CaseRule *cas : sw->cases)
+		for (const RTLIL::SigSpec &sig : cas->compare)
+			if (!sig.is_fully_def())
+				return false;
+
+	return true;
+}
+
+// This replicates the necessary parts of BitPatternPool's interface, but rather
+// than storing remaining patterns, this explicitly stores which fully-defined
+// constants have already been matched.
+struct ExplicitConstPool
+{
+	ExplicitConstPool(const RTLIL::SigSpec &signal)
+		: max_patterns{signal.size() >= 64 ? 0ul : 1ul << signal.size()}
```

Comment:
`1ul << signal.size()` breaks on 32-bit platforms for eg. 48-bit signals

Reply:
I'd say limit this to 31-bit signals, any switch that saturates a 32-bit signal is way too big to be processed by yosys anyway